pepper-apis/src/main/java/org/broadinstitute/ddp/model/event/activityinstancecreation/ActivityInstanceCreatorFromAnswers.java

```diff
@@ -0,0 +1,165 @@
+package org.broadinstitute.ddp.model.event.activityinstancecreation;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.broadinstitute.ddp.db.DaoException;
+import org.broadinstitute.ddp.db.dao.AnswerDao;
+import org.broadinstitute.ddp.db.dao.JdbiActivity;
+import org.broadinstitute.ddp.db.dao.JdbiActivityInstance;
+import org.broadinstitute.ddp.db.dao.JdbiActivityInstanceStatus;
+import org.broadinstitute.ddp.db.dao.JdbiQuestionCached;
+import org.broadinstitute.ddp.db.dto.ActivityDto;
+import org.broadinstitute.ddp.db.dto.QuestionDto;
+import org.broadinstitute.ddp.exception.DDPException;
+import org.broadinstitute.ddp.model.activity.instance.answer.Answer;
+import org.broadinstitute.ddp.model.activity.instance.answer.CompositeAnswer;
+import org.broadinstitute.ddp.model.activity.instance.answer.PicklistAnswer;
+import org.broadinstitute.ddp.model.event.EventSignal;
+import org.broadinstitute.ddp.service.ActivityInstanceCreationService;
+import org.jdbi.v3.core.Handle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Create  activity instances from the selected answers of parent instances.
+ * Question stable_id which answers to check is specified by parameter {@link #sourceQuestionStableId}.
+ *
+ * <p><b>Algorithm:</b>
+ * <ul>
+ *     <li>find answer (picklist) by 'sourceQuestionStableId' and detect selected options;</li>
+ *     <li>for each of selected options:</li>
+ *     <li> - create a new instance (of activity with ID='studyActivityId';</li>
+ *     <li> - if `targetQuestionStableId` is specified then copy source answers to target answers;</li>
+ * </ul>
+ */
+public class ActivityInstanceCreatorFromAnswers extends ActivityInstanceCreatorDefault {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ActivityInstanceCreatorFromAnswers.class);
+
+    private final String sourceQuestionStableId;
+    private final String targetQuestionStableId;
+
+    public ActivityInstanceCreatorFromAnswers(
+            long studyActivityId,
+            String sourceQuestionStableId,
+            String targetQuestionStableId,
+            ActivityInstanceCreationService creationService) {
+        super(studyActivityId, creationService);
+        this.sourceQuestionStableId = sourceQuestionStableId;
+        this.targetQuestionStableId = targetQuestionStableId;
+    }
+
+    /**
+     * Create N activity instances.
+     * where N = number of answers in composite question `sourceQuestionStableId`.
+     * IF N > not_used_activities_count THEN N = not_used_activities_count,
```

Comment:
I think for now, it might be better if we throw an exception when it's `N > not_used` ? If we're using this feature, then I'm pretty sure we're expecting to create all N, so if we can't then let's give an error so we can further investigate.

Reply:
Fixed